payjoin/src/core/send/v2/mod.rs

```diff
@@ -131,11 +114,26 @@ impl<'a> SenderBuilder<'a> {
         self,
         min_fee_rate: FeeRate,
     ) -> MaybeBadInitInputsTransition<SessionEvent, Sender<WithReplyKey>, BuildSenderError> {
-        let v1 = match self.0.build_non_incentivizing(min_fee_rate) {
+        self.v2_sender_from_v1(self.0.clone().build_non_incentivizing(min_fee_rate))
```

Comment:
These `self.0.clone()` smell like we're taking on tech debt but I couldn't find an easy solution in my 15 minutes of play with it.